src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -1794,6 +1804,31 @@ class BookingService(
 
     return Triple(applicationId, eventNumber, submittedAt)
   }
+
+  @SuppressWarnings("ThrowsCount", "TooGenericExceptionCaught")
+  private fun closeTransitionalAccommodationAssessment(
+    assessmentId: UUID,
+    user: UserEntity,
+    booking: BookingEntity,
+  ) {
+    try {
+      val assessmentValidationResult = when (val assessmentAuthResult = assessmentService.closeAssessment(user, assessmentId)) {
+        is AuthorisableActionResult.Success -> assessmentAuthResult.entity
+        is AuthorisableActionResult.NotFound -> throw NotFoundProblem(assessmentId, "Assessment")
+        is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
+      }
+
+      when (assessmentValidationResult) {
+        is ValidatableActionResult.GeneralValidationError -> throw BadRequestProblem(errorDetail = assessmentValidationResult.message)
+        is ValidatableActionResult.FieldValidationError -> throw BadRequestProblem(invalidParams = assessmentValidationResult.validationMessages)
+        is ValidatableActionResult.ConflictError -> throw ConflictProblem(id = assessmentValidationResult.conflictingEntityId, conflictReason = assessmentValidationResult.message)
+        is ValidatableActionResult.Success -> Unit
+      }
+    } catch (exception: Exception) {
```

Comment:
would changing this to catch ThrowableProblem work? (trying to get rid of the TooGenericExceptionCaught warning)

I don't think we should be capturing RuntimeExceptions (e.g. database issues), IMO these should really propagate back to the caller because if they retry it _should_ work (instead of leaving the assessment in an unexpected state)

Reply:
As i mentioned in the PR comments, we dont want to fail the booking when if we cant close the assessment and that the reason we capture all exception and log the response.

Reply:
I agree with the end goal, but I don't think we should be capturing high level exceptions (like Exception or RuntimeException) because this will capturing things we probably should let propagate up (e.g. database errors, coding errors etc.). I think it's fine to capture something more specific e.g. ThrowableProblem and we can then also remove the Detekt suppression. I'll leave conversation unresolved by give approval for merge, up to you if you want to change this 👍

Reply:
I don't think we should propagate the error to high level in this case because:

1) Error while closing the assessment is not fatal error which we need to propagate and fail the transaction. The code is catching the exception and it knows what to do with it. You need to propagate to next level only when you dont know what to do with it.

2) We anyway catch the exception and raise alert if we get unexpected error for support team to investigate the issue.So we are raising  exception for investigation and not swolling it.

3) The main requirement is to try close the assessment automatically ,if we cant close it automatically then dont fail the booking. We have alternative option flow for the user to manually close the assessment .